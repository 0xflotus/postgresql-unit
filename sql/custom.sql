SELECT '1 foobar'::unit;
SELECT unit_is_hashed('foobar');
INSERT INTO unit_prefixes VALUES ('foo', 42);
SELECT '1 foobar'::unit;
SELECT unit_is_hashed('foobar');
SELECT * FROM unit_prefixes WHERE dump;

INSERT INTO unit_units VALUES ('legobrick', '9.6 mm');
SELECT unit_is_hashed('legobricks');
SELECT '1 m'::unit @ 'legobricks' AS one_meter;
SELECT unit_is_hashed('legobricks');
SELECT * FROM unit_units WHERE dump;

UPDATE unit_units SET unit = '19.1 mm' WHERE name = 'legobrick'; -- Duplo size
SELECT '2 legobricks'::unit AS old_size;
SELECT unit_reset();
SELECT unit_is_hashed('legobricks');
SELECT '2 legobricks'::unit AS new_size;
