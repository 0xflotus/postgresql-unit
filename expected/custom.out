SELECT '1 foobar'::unit;
ERROR:  unit "foobar" is not known
LINE 1: SELECT '1 foobar'::unit;
               ^
SELECT unit_is_hashed('foobar');
 unit_is_hashed 
----------------
 f
(1 row)

INSERT INTO unit_prefixes VALUES ('foo', 42);
SELECT '1 foobar'::unit;
  unit   
---------
 4.2 MPa
(1 row)

SELECT unit_is_hashed('foobar');
 unit_is_hashed 
----------------
 t
(1 row)

SELECT * FROM unit_prefixes WHERE dump;
 prefix | factor | definition | dump 
--------+--------+------------+------
 foo    |     42 |            | t
(1 row)

INSERT INTO unit_units VALUES ('legobrick', '9.6 mm');
SELECT unit_is_hashed('legobricks');
 unit_is_hashed 
----------------
 f
(1 row)

SELECT '1 m'::unit @ 'legobricks' AS one_meter;
          one_meter          
-----------------------------
 104.166666666667 legobricks
(1 row)

SELECT unit_is_hashed('legobricks');
 unit_is_hashed 
----------------
 t
(1 row)

SELECT * FROM unit_units WHERE dump;
   name    |  unit  | shift | definition | dump 
-----------+--------+-------+------------+------
 legobrick | 9.6 mm |       |            | t
(1 row)

